worker/jobTypes/githubJob.js

```diff
@@ -0,0 +1,114 @@
+const fs = require('fs-extra');
+const workerUtils = require('../utils/utils');
+const simpleGit = require('simple-git/promise');
+
+class GitHubJobClass {
+
+  // pass in a job payload to setup class
+  constructor(currentJob) {
+    this.currentJob = currentJob;
+  }
+
+  // cleanup before pulling repo
+  async cleanup(logger) {
+    const currentJob = this.currentJob;
+    try {
+      workerUtils.removeDirectory(`repos/${workerUtils.getRepoDirName(currentJob)}`);
+    } catch (errResult) {
+      logger.save(`${'(CLEANUP)'.padEnd(15)}failed cleaning repo directory`);
+      throw errResult;
+    }
+    return new Promise(function(resolve, reject) {
+      resolve(true);
+    });
+  }
+
+  async cloneRepo(logger) {
+    const currentJob = this.currentJob;
+    try {
+      if (!currentJob.payload.branchName) {
+        logger.save(`${'(CLONE)'.padEnd(15)}failed due to insufficient definition`);
+        throw new Error('branch name not indicated');
+      }
+      const basePath = workerUtils.getBasePath(currentJob);
+      const repoPath = basePath + '/' + currentJob.payload.repoOwner + '/' + currentJob.payload.repoName;
+      await simpleGit('repos')
+        .silent(false)
+        .clone(repoPath, `${workerUtils.getRepoDirName(currentJob)}`)
+        .catch(err => {
+          console.error('failed: ', err);
+          throw err;
+        });
+    } catch (errResult) {
+      if (
+        errResult.hasOwnProperty('code') ||
+        errResult.hasOwnProperty('signal') ||
+        errResult.hasOwnProperty('killed')
+      ) {
+        logger.save(`${'(GIT)'.padEnd(15)}failed with code: ${errResult.code}`);
+        logger.save(`${'(GIT)'.padEnd(15)}stdErr: ${errResult.stderr}`);
+        throw errResult;
+      }
+    }
+    return new Promise(function(resolve, reject) {
+      resolve(true);
+    });
+  }
+
+  async buildRepo(logger) {
+    const currentJob = this.currentJob;
```

Comment:
one further abstraction would be to have buildRepo call cleanup and clone and pull that out of the githubPushJob.js to make it even cleaner.